perceval/backends/mozilla_reps.py

```diff
@@ -0,0 +1,263 @@
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaReps item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return MozillaReps.__metadata_updated_on_event(item)
+
+    @staticmethod
+    def __metadata_id_event(item):
+        return str(item['event_url'])
+
+    @staticmethod
+    def __metadata_updated_on_event(item):
+        return float(parser.parse(item['end']).timestamp())
+
+
+class MozillaRepsClient:
+    """MozillaReps API client.
+
+    This class implements a simple client to retrieve builds from
+    projects in a MozillaReps node.
+
+    :param url: URL of MozillaReps url https://reps.mozilla.org
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    MAX_ITEMS_EV = "40"  # max number of events per iteration
+
+    def __init__(self, url):
+        self.url = url
+        self.api_events_url = urljoin(self.url, '/api/v1/event/')
+        self.api_events_url += '/'  # API needs a final /
+
+    def call(self, params):
+        """Run an API command.
+        :param cgi: cgi command to run on the server
+        :param params: dict with the HTTP parameters needed to run
+            the given command
+        """
+        logger.debug("MozillaReps client calls API: %s params: %s",
+                     self.api_events_url, str(params))
+
+        req = requests.get(self.api_events_url, params=params)
+        req.raise_for_status()
+
+        return req.text
+
+    def get_events(self,  next_url=None):
+        """Retrieve all events"""
+        if not next_url:
+            next_url = "/api/v1/event/?offset=0&limit=%s" % self.MAX_ITEMS_EV
+
+        params = {
+            "offset":"0",
+            "limit":self.MAX_ITEMS_EV
+        }
+
+        return self.call(params)
+
+
+class MozillaRepsCommand(BackendCommand):
+    """Class to run MozillaReps backend from the command line."""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self.url = self.parsed_args.url
+        self.origin = self.parsed_args.origin
+        self.outfile = self.parsed_args.outfile
+
+        if not self.parsed_args.no_cache:
+            if not self.parsed_args.cache_path:
+                base_path = os.path.expanduser('~/.perceval/cache/')
+            else:
+                base_path = self.parsed_args.cache_path
+
+            cache_path = os.path.join(base_path, self.url)
+
+            cache = Cache(cache_path)
+
+            if self.parsed_args.clean_cache:
+                cache.clean()
+            else:
+                cache.backup()
+        else:
+            cache = None
+
+        self.backend = MozillaReps(self.url, cache=cache, origin=self.origin)
+
+    def run(self):
+        """Fetch and print the Builds.
```

Comment:
Arrgg, I think I have removed all references to builds (from jenkins backend) but it seems I have not done it totally. Reviewing ... done! The case sensitive replacement was the problem.